### PR TITLE
Bump Katello to 4.0.0

### PR DESCRIPTION
**Pull request by jturel**

### CODE REVIEW

**Comment by jturel** [comment]
@jlsherrill can you let me know if I should bump any of the pulp dependencies while I'm at it?

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 1]
```diff
@@ -8,8 +8,8 @@
 %global gem_name katello
 %global prereleasesource pre.master
 %global prerelease %{?prereleasesource:.}%{?prereleasesource}
-%global mainver 3.18.0
-%global release 5
+%global mainver 4.0.0
```
Comment (ekohl) [inline packages/katello/rubygem-katello/rubygem-katello.spec:11]: Please also update foreman_{min,max}_version in this file

**Review by ekohl — CHANGES_REQUESTED** [review]

File: packages/katello/rubygem-katello/rubygem-katello.spec (hunk 1 above)
Comment (jturel) [reply packages/katello/rubygem-katello/rubygem-katello.spec:11]: Updated. Thanks for the catch

**Comment by ekohl** [comment]
https://ci.theforeman.org/job/katello-master-source-release/ still produced a 3.18.0 gem. https://github.com/Katello/katello/commit/2d977a4d090269cf456da2c683b848ffd24c090d was merged, but doesn't look like there's a build for that.

**Comment by tbrisker** [comment]
[test rpm]

**Comment by ekohl** [comment]
Looks like those new dependencies are needed:
```
Gem loading error: Could not find 'pulp_deb_client' (< 2.8.0, >= 2.6.0) among 167 total gem(s)
```

**Review by ekohl — CHANGES_REQUESTED** [review]
You missed the `BuildRequires` which means it still isn't present during the build.

**Comment by jturel** [comment]
> You missed the `BuildRequires` which means it still isn't present during the build.

oh my gosh :)

**Review by ekohl — APPROVED** [review]
When this is green, I think it's good to merge.

**Review by evgeni — CHANGES_REQUESTED** [review]
Foreman min/max version need adjustment, as develop is 2.4 now ;)

**Comment by ekohl** [comment]
> Foreman min/max version need adjustment, as develop is 2.4 now ;)

That's done in rubygem-katello.spec. I'm not aware of another place to do so.

**Comment by jturel** [comment]
I think the foreman versions I have in this PR are accurate:

```
%global foreman_min_version 2.4
%global foreman_max_version 2.5
```

**Review by evgeni — APPROVED** [review]
repoclosure output tricked me!

**Comment by evgeni** [comment]
uh, but, you also want to bump katello-repos :see_no_evil:

**Comment by jturel** [comment]
> uh, but, you also want to bump katello-repos see_no_evil

truuuuuue!

**Review by evgeni — APPROVED** [review]

**Pull request merged**